From 661726bb8dd19d906d96d75a1c0a3b299ccd2994 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 12 Dec 2024 14:10:57 +0800
Subject: [PATCH 287/301] AOSCOS: MIPS: loongson3-memcpy: adapt to
 RAW_COPY_USER

Since commit 2260ea86c0e7 ("mips: switch to RAW_COPY_USER"), it is
required to export `__raw_copy_from_user' and `__raw_copy_from_user',
which are then referenced by certain kernel modules.

Export these symbols under `__memcpy', also remove the export for the now-
unused `__copy_user' symbol.

Fixes: "AOSCOS: memcpy optimization for loongson-3"
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 arch/mips/loongson64/loongson3-memcpy.S | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/mips/loongson64/loongson3-memcpy.S b/arch/mips/loongson64/loongson3-memcpy.S
index 2021e75d5dd2..c5dd2e2866c3 100644
--- a/arch/mips/loongson64/loongson3-memcpy.S
+++ b/arch/mips/loongson64/loongson3-memcpy.S
@@ -119,8 +119,10 @@ LEAF(memcpy)				/* a0=dst a1=src a2=len */
 EXPORT_SYMBOL(memcpy)
 	move	v0, dst
 .L__memcpy:
-FEXPORT(__copy_user)
-EXPORT_SYMBOL(__copy_user)
+FEXPORT(__raw_copy_from_user)
+EXPORT_SYMBOL(__raw_copy_from_user)
+FEXPORT(__raw_copy_to_user)
+EXPORT_SYMBOL(__raw_copy_to_user)
 	/* if less then 0x28 bytes */
 	sltu	t2, a2, 0x28
 	andi	t0, dst, 0xf
-- 
2.48.1

